addons/hw_drivers/iot_handlers/interfaces/PrinterInterface.py

```diff
@@ -22,16 +22,24 @@ def get_devices(self):
             devices = conn.getDevices()
             for printer in printers:
                 path = printers.get(printer).get('device-uri', False)
-                if path and path in devices:
-                    devices.get(path).update({'supported': True}) # these printers are automatically supported
+                if printer != self.get_identifier(path):
```

Comment:
in this function, why do you need printers.get(printer)? what "printers" is?

Can't we do "for printer_id,printer in printers?

Reply:
`printers` is a `dict` returned by cups function `getPrinters()` like `getDevices()` but printers are device already setted in cups

here printer is the `key` and `printers` the `dict`
sample:

```
.
.
.
    "test_ipp": {
        "printer-is-shared": false,
        "printer-state": 3,
        "printer-state-message": "",
        "printer-state-reasons": [
            "none"
        ],
        "printer-type": 2097158,
        "printer-uri-supported": "ipp://localhost/printers/test_ipp",
        "printer-location": "",
        "printer-info": "test_ipp",
        "device-uri": "ipp://localhost/printers/socket192168125",
        "printer-make-and-model": "Remote Printer"
    },
    "wdfgdfg": {
        "printer-is-shared": false,
        "printer-state": 3,
        "printer-state-message": "",
        "printer-state-reasons": [
            "none"
        ],
        "printer-type": 2228228,
        "printer-uri-supported": "ipp://localhost/printers/wdfgdfg",
        "printer-location": "",
        "printer-info": "HP OfficeJet 8010 series",
        "device-uri": "dnssd://HP%20OfficeJet%208010%20series%20%5B463ED4%5D._ip                                                                            p._tcp.local/?uuid=81831c7b-cf58-57f3-60b7-baf05417fde0",
        "printer-make-and-model": "Local Raw Printer"
    }
}
```

Reply:
ok, vu que c'est un dic, tu peux modifier ton for en "for id, printer in printers.items():", et directement utiliser printer (printer sera égal à printers.get(printer))